application/apps/rustcore/rs-bindings/src/js/jobs/mod.rs

```diff
@@ -82,53 +99,146 @@ impl UnboundJobs {
 
     /// Cancel given operation/task
     #[node_bindgen]
-    async fn abort(&self, operation_uuid: String) -> Result<(), ComputationErrorWrapper> {
+    async fn abort(&self, id: i64) -> Result<(), ComputationErrorWrapper> {
         self.api
             .as_ref()
             .ok_or(ComputationError::SessionUnavailable)?
-            .cancel_job(
-                &operations::uuid_from_str(&operation_uuid).map_err(ComputationErrorWrapper)?,
-            )
+            .cancel_job(&id_from_i64(id)?)
             .await
             .map_err(ComputationErrorWrapper)
     }
 
     // Custom methods (jobs)
     #[node_bindgen]
-    async fn list_folder_content<F: Fn(String) + Send + 'static>(
+    async fn list_folder_content(
+        &self,
+        id: i64,
+        depth: i64,
+        path: String,
+    ) -> Result<CommandOutcomeWrapper<String>, ComputationErrorWrapper> {
+        self.api
+            .as_ref()
+            .ok_or(ComputationError::SessionUnavailable)?
+            .list_folder_content(id_from_i64(id)?, usize_from_i64(depth)?, path)
+            .await
+            .map_err(ComputationErrorWrapper)
+            .map(CommandOutcomeWrapper)
+    }
+
+    #[node_bindgen]
+    async fn spawn_process(
```

Comment:
2 things:
* ID should come from rust core
* name "spawn_process" is expressing implementation details

Reply:
> ID should come from rust core
that's the cost of refusing callbacks. We are sending id into rustcore from js and do not need callback, which was used to delivery id from rustcore. Put your attention, we are safe from a situation of id duplicate. In case Id already exists - we will return an error.

Reply:
> name "spawn_process" is expressing implementation details

This command just spawn an external command/program. We are already refusing usage of it, not because it doesn't work, but because it works from a binding context and gives some issues. But it wasn't removed yet, as soon as I didn't investigate yet, do we still places where it could be useful or not. Before it was used to run updater, but for updater it does't work because of binding context. For other stuff it's still could be useful